src/analyses/README.md

```diff
@@ -121,6 +121,144 @@ is covered in a separate subsection.
 
 A basic description for how a natural loop works is here: https://web.cs.wpi.edu/~kal/PLT/PLT8.6.4.html
 
+### Reaching definitions
+
+This module implements the reaching definitions data-flow analysis
+(see https://en.wikipedia.org/wiki/Reaching_definition for basic introduction)
+on multi-threaded GOTO programs.
+
+The domain element is defined in a class `rd_range_domaint` and the actual
+analysis is implemented in a class `reaching_definitions_analysist`. Beside
+these classes there are more data structures necessary in the computation.
+We discuss all of them in the following sub-sections.   
+
+#### Struct `reaching_definitiont`
```

Comment:
🚫I think this detailed documentation should be moved to the source and header files. Only a conceptual overview with refs to the code should be given here.

Reply:
I could certainly do that. Perhaps in the next doc day?

Reply:
It's probably half an hour work to copy the text into the right places. I'd prefer it to be done now.

Reply:
Done.